File: cdm-sources/cdm/product/collateral/functions/CheckIssuerName.java
package cdm.product.collateral.functions;

import cdm.base.staticdata.party.LegalEntity;
import cdm.product.collateral.EligibilityQuery;
import cdm.product.collateral.IssuerName;
import com.google.inject.ImplementedBy;
import com.rosetta.model.lib.functions.RosettaFunction;
import com.rosetta.model.lib.mapper.MapperS;

import static com.rosetta.model.lib.expression.ExpressionOperators.*;

@ImplementedBy(CheckIssuerName.CheckIssuerNameDefault.class)
public abstract class CheckIssuerName implements RosettaFunction {

	/**
	* @param issuerName 
	* @param query 
	* @return isEqual 
	*/
	public Boolean evaluate(IssuerName issuerName, EligibilityQuery query) {
		Boolean isEqual = doEvaluate(issuerName, query);
		
		return isEqual;
	}

	protected abstract Boolean doEvaluate(IssuerName issuerName, EligibilityQuery query);

	public static class CheckIssuerNameDefault extends CheckIssuerName {
		@Override
		protected Boolean doEvaluate(IssuerName issuerName, EligibilityQuery query) {
			Boolean isEqual = null;
			return assignOutput(isEqual, issuerName, query);
		}
		
		protected Boolean assignOutput(Boolean isEqual, IssuerName issuerName, EligibilityQuery query) {
			isEqual = notExists(MapperS.of(issuerName)).or(contains(MapperS.of(issuerName).<LegalEntity>map("getIssuerName", _issuerName -> _issuerName.getIssuerName()), MapperS.of(query).<LegalEntity>map("getIssuerName", eligibilityQuery -> eligibilityQuery.getIssuerName()))).get();
			
			return isEqual;
		}
	}
}

File: cdm-sources/src/cdm/product/collateral/functions/CheckIssuerName.py
# pylint: disable=line-too-long, invalid-name, missing-function-docstring, missing-module-docstring, superfluous-parens
# pylint: disable=wrong-import-position, unused-import, unused-wildcard-import, wildcard-import, wrong-import-order, missing-class-docstring
from __future__ import annotations
import sys
import datetime
import inspect
from decimal import Decimal
from rosetta.runtime.utils import *
from rosetta.runtime.func_proxy import replaceable, create_module_attr_guardian
from cdm.product.collateral.EligibilityQuery import EligibilityQuery
from cdm.product.collateral.IssuerName import IssuerName

__all__ = ['CheckIssuerName']


@replaceable
def CheckIssuerName(issuerName: IssuerName | None, query: EligibilityQuery) -> bool:
    """
    
    Parameters 
    ----------
    issuerName : IssuerName
    
    query : EligibilityQuery
    
    Returns
    -------
    isEqual : boolean
    
    """
    self = inspect.currentframe()
    
    
    isEqual =  ((not rosetta_attr_exists(rosetta_resolve_attr(self, "issuerName"))) or contains(rosetta_resolve_attr(rosetta_resolve_attr(self, "issuerName"), "issuerName"), rosetta_resolve_attr(rosetta_resolve_attr(self, "query"), "issuerName")))
    
    
    return isEqual

sys.modules[__name__].__class__ = create_module_attr_guardian(sys.modules[__name__].__class__)
